pkg/apic/specoas3processor.go

```diff
@@ -140,3 +146,37 @@ func (p *oas3SpecProcessor) parseURLsIntoEndpoints(defaultURL string, allURLs []
 
 	return endPoints, nil
 }
+
+// SetServers replaces the servers array on the Openapi3 object.
+func SetServers(hosts []string, spec *openapi3.Swagger) {
+	var oas3Servers []*openapi3.Server
+	for _, s := range hosts {
+		oas3Servers = append(oas3Servers, &openapi3.Server{
+			URL: s,
+		})
+	}
+	if len(oas3Servers) > 0 {
+		spec.Servers = oas3Servers
+	}
+}
+
+// ParseOAS3 converts a JSON or YAML spec into an OpenAPI3 object
+func ParseOAS3(spec []byte) (*openapi3.Swagger, error) {
+	oas3Obj, err := openapi3.NewSwaggerLoader().LoadSwaggerFromData(spec)
+	if err != nil {
+		return nil, err
+	}
+	if !strings.Contains(oas3Obj.OpenAPI, "3.") {
+		return nil, fmt.Errorf("Invalid openapi 3 specification. 'openapi' must be version '3.0' or above.")
+	}
+	if oas3Obj.Paths == nil {
+		return nil, fmt.Errorf("Invalid openapi 3 specification. 'paths' key not found.")
+	}
+	if oas3Obj.Info == nil {
+		return nil, fmt.Errorf("Invalid openapi 3 specification. 'info' key not found.")
+	}
+	if oas3Obj.Info.Title == "" {
+		return nil, fmt.Errorf("Invalid openapi 3 specification. 'info.title' key not found.")
```

Comment:
aside from the linting error, these 4 error messages are mostly the same, can they be combined into a single string and with string substitution?